## Import Essential Libraries and Load in Data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Compiling Approved Loan Data

In [2]:
# 2007 - 2013
df_approv_07_11 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2007-2011csv/LoanStats3a.csv', header=1)
df_approv_12_13 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2012-2013csv/LoanStats3b.csv', header=1)

C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,47) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,47,123,124,125,128,129,130,133) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [12]:
# 2014-2015
df_approv_14 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2014.csv/LoanStats3c.csv', header=1)
df_approv_15 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2015.csv/LoanStats3d.csv', header=1)

C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,19) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,19,55) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [23]:
# 2016 by quarter
df_approv_16q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2016Q1.csv/LoanStats_2016Q1.csv', header=1)
df_approv_16q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2016Q2.csv/LoanStats_2016Q2.csv', header=1)
df_approv_16q3 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2016Q3.csv/LoanStats_2016Q3.csv', header=1)
df_approv_16q4 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2016Q4.csv/LoanStats_2016Q4.csv', header=1)

C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,19) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,19,55) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [72]:
# 2016
df_approv_16 = pd.concat([df_approv_16q1,df_approv_16q2,df_approv_16q3,df_approv_16q4])

In [34]:
# 2017 by quarter
df_approv_17q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2017Q1.csv/LoanStats_2017Q1.csv', header=1)
df_approv_17q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2017Q2.csv/LoanStats_2017Q2.csv', header=1)
df_approv_17q3 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2017Q3.csv/LoanStats_2017Q3.csv', header=1)
df_approv_17q4 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2017Q4.csv/LoanStats_2017Q4.csv', header=1)

C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,112) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,123,124,125,128,129,130,133,139,140,141) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [73]:
# 2017
df_approv_17 = pd.concat([df_approv_17q1,df_approv_17q2,df_approv_17q3,df_approv_17q4])

In [38]:
# 2018 first two quarters
df_approv_18q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2018Q1.csv/LoanStats_2018Q1.csv', header=1)
df_approv_18q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/2018Q2.csv/LoanStats_2018Q2.csv', header=1)

C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,123,124,125,128,129,130,133,139,140,141) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\josh\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2698: DtypeWarning: Columns (0,139,140,141) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [74]:
# 2018
df_approv_18 = pd.concat([df_approv_18q1,df_approv_18q2])

Before we compile everything into one, let's examine their shape and on what columns we can merge the dataframes

In [75]:
df_approv_07_11.shape, df_approv_12_13.shape, df_approv_14.shape, df_approv_15.shape, df_approv_16.shape,\
df_approv_17.shape, df_approv_18.shape

((42538, 145),
 (188183, 145),
 (235631, 145),
 (421097, 145),
 (434415, 145),
 (443587, 145),
 (238640, 145))

All dataframes seem to have the same number of columns. Let's check if all of the columns are the same ones throughout all dataframes

In [43]:
(df_approv_07_11.columns == df_approv_12_13.columns).all()

True

In [44]:
(df_approv_12_13.columns == df_approv_14.columns).all()

True

In [45]:
(df_approv_14.columns == df_approv_15.columns).all()

True

In [76]:
(df_approv_15.columns == df_approv_16.columns).all()

True

In [77]:
(df_approv_17.columns == df_approv_18.columns).all()

True

They are! So we can easily concatenate them

In [78]:
df_approv = pd.concat([df_approv_07_11,df_approv_12_13,df_approv_14,df_approv_15,df_approv_16,df_approv_17,df_approv_18])

In [79]:
df_approv.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2004091 entries, 0 to 130773
Columns: 145 entries, id to settlement_term
dtypes: float64(106), object(39)
memory usage: 2.2+ GB


In [80]:
df_approv.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  NaN        NaN     5000.0       5000.0           4975.0   36 months   
1  NaN        NaN     2500.0       2500.0           2500.0   60 months   
2  NaN        NaN     2400.0       2400.0           2400.0   36 months   
3  NaN        NaN    10000.0      10000.0          10000.0   36 months   
4  NaN        NaN     3000.0       3000.0           3000.0   60 months   

  int_rate  installment grade sub_grade       ...        \
0   10.65%       162.87     B        B2       ...         
1   15.27%        59.83     C        C4       ...         
2   15.96%        84.33     C        C5       ...         
3   13.49%       339.31     C        C1       ...         
4   12.69%        67.79     B        B5       ...         

  hardship_payoff_balance_amount hardship_last_payment_amount  \
0                            NaN                          NaN   
1                            NaN                          NaN   
2                            NaN                          NaN   
3                            NaN                          NaN   
4                            NaN                          NaN   

  disbursement_method  debt_settlement_flag debt_settlement_flag_date  \
0                Cash                     N                       NaN   
1                Cash                     N                       NaN   
2                Cash                     N                       NaN   
3                Cash                     N                       NaN   
4                Cash                     N                       NaN   

  settlement_status settlement_date settlement_amount  settlement_percentage  \
0               NaN             NaN               NaN                    NaN   
1               NaN             NaN               NaN                    NaN   
2               NaN             NaN               NaN                    NaN   
3               NaN             NaN               NaN                    NaN   
4               NaN             NaN               NaN                    NaN   

  settlement_term  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  

[5 rows x 145 columns]

In [84]:
df_approv.to_csv('E:/josh_data_science/Lending Club Loan Data/Approved Loan Data/approved_loan_data.csv')

## Compiling Rejected Loan Data

In [86]:
# 2007 - 2014
df_dec_07_12 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2007-2012.csv/RejectStatsA.csv', header=1)
df_dec_13_14 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2013-2014.csv/RejectStatsB.csv', header=1)

In [87]:
# 2015
df_dec_15 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2015csv/RejectStatsD.csv', header=1)

In [88]:
# 2016 by quarter
df_dec_16q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2016Q1.csv/RejectStats_2016Q1.csv', header=1)
df_dec_16q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2016Q2.csv/RejectStats_2016Q2.csv', header=1)
df_dec_16q3 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2016Q3.csv/RejectStats_2016Q3.csv', header=1)
df_dec_16q4 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2016Q4.csv/RejectStats_2016Q4.csv', header=1)

In [90]:
# 2016
df_dec_16 = pd.concat([df_dec_16q1,df_dec_16q2,df_dec_16q3,df_dec_16q4])

In [91]:
# 2017 by quarter
df_dec_17q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2017Q1.csv/RejectStats_2017Q1.csv', header=1)
df_dec_17q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2017Q2.csv/RejectStats_2017Q2.csv', header=1)
df_dec_17q3 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2017Q3.csv/RejectStats_2017Q3.csv', header=1)
df_dec_17q4 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2017Q4.csv/RejectStats_2017Q4.csv', header=1)

In [92]:
# 2017
df_dec_17 = pd.concat([df_dec_17q1,df_dec_17q2,df_dec_17q3,df_dec_17q4])

In [94]:
# 2018 first two quarters
df_dec_18q1 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2018Q1.csv/RejectStats_2018Q1.csv', header=1)
df_dec_18q2 = pd.read_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/2018Q2.csv/RejectStats_2018Q2.csv', header=1)

In [97]:
# 2018
df_dec_18 = pd.concat([df_dec_18q1,df_dec_18q2])

Let's check if all the dataframes for all years have the same dimension

In [98]:
df_dec_07_12.shape, df_dec_13_14.shape, df_dec_15.shape, df_dec_16.shape, df_dec_17.shape, df_dec_18.shape

((755491, 9),
 (2694642, 9),
 (2859379, 9),
 (4769874, 9),
 (7072573, 9),
 (4317115, 9))

They do! Are all 9 columns the same throughout all dataframes?

In [99]:
(df_dec_07_12.columns==df_dec_13_14.columns).all()

True

In [100]:
(df_dec_13_14.columns==df_dec_15.columns).all()

True

In [101]:
(df_dec_15.columns==df_dec_16.columns).all()

True

In [102]:
(df_dec_16.columns==df_dec_17.columns).all()

True

In [103]:
(df_dec_17.columns==df_dec_18.columns).all()

True

They are! So we can easily concatenate them just like for the approved loan data case

In [104]:
df_declined = pd.concat([df_dec_07_12,df_dec_13_14,df_dec_15,df_dec_16,df_dec_17,df_dec_18])

In [105]:
df_declined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 22469074 entries, 0 to 2441980
Data columns (total 9 columns):
Amount Requested        float64
Application Date        object
Loan Title              object
Risk_Score              float64
Debt-To-Income Ratio    object
Zip Code                object
State                   object
Employment Length       object
Policy Code             float64
dtypes: float64(3), object(6)
memory usage: 1.7+ GB


In [106]:
df_declined.head()

Amount Requested Application Date                        Loan Title  \
0            1000.0       2007-05-26  Wedding Covered but No Honeymoon   
1            1000.0       2007-05-26                Consolidating Debt   
2           11000.0       2007-05-27       Want to consolidate my debt   
3            6000.0       2007-05-27                           waksman   
4            1500.0       2007-05-27                            mdrigo   

   Risk_Score Debt-To-Income Ratio Zip Code State Employment Length  \
0       693.0                  10%    481xx    NM           4 years   
1       703.0                  10%    010xx    MA          < 1 year   
2       715.0                  10%    212xx    MD            1 year   
3       698.0               38.64%    017xx    MA          < 1 year   
4       509.0                9.43%    209xx    MD          < 1 year   

   Policy Code  
0          0.0  
1          0.0  
2          0.0  
3          0.0  
4          0.0

In [107]:
df_declined.to_csv('E:/josh_data_science/Lending Club Loan Data/Declined Loan Data/declined_loan_data.csv')

**To summarize, approved loan data had 2004091 observations and 145 variables while the declined loan data had 22469074 observations and 9 variables.**